drgn/helpers/linux/kconfig.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+Kconfig
+------
+
+
+The ``drgn.helpers.linux.kconfig`` module provides helpers for reading kernel
+configuration from memory. This is only supported if CONFIG_IKCONFIG is set.
+"""
+
+import gzip
+import types
+
+__all__ = (
+    "get_kconfig",
+)
+
+def get_kconfig(prog):
+    try:
+        return prog.cache["kconfig_options_map"]
+    except KeyError:
+        pass
+
+    try:
+        start = prog.symbol("kernel_config_data").address
+        size = prog.symbol("kernel_config_data_end").address - start
+        raw_data = prog.read(start, size)
+    except LookupError:
+        try:
+            # Try reading data in format present before kernel 5.0.
+            start = prog['kernel_config_data'].address_
+            size = len(prog['kernel_config_data'])
+            raw_data = prog.read(start, size)
+            trim_left = raw_data.find(b'IKCFG_ST') + len(b'IKCFG_ST')
+            trim_right = raw_data.rfind(b'IKCFG_ED')
+            if trim_left == -1 or trim_right == -1:
```

Comment:
I don't think we need to actually find the delimiters; they're guaranteed to be at the start and end of the array.